pkg/pillar/cmd/client/client.go

```diff
@@ -819,8 +819,8 @@ func handleDNSImpl(ctxArg interface{}, key string,
 		return
 	}
 	log.Functionf("handleDNSImpl for %s", key)
-	// Ignore test status and timestamps
-	if ctx.deviceNetworkStatus.MostlyEqual(status) {
+	// Ignore timestamps
+	if status.State == ctx.networkState && ctx.deviceNetworkStatus.MostlyEqual(status) {
```

Comment:
Based on the comments for MostlyEqual in zedroutertypes.go it might make sense to put this check inside that function. I suspect but I haven't checked, that other callers would also want to react to the change in the top-level State, However, here you are comparing ctx.networkState and not ctx.deviceNetworkStatus.State - why is that?

Reply:
> Based on the comments for MostlyEqual in zedroutertypes.go it might make sense to put this check inside that function. I suspect but I haven't checked, that other callers would also want to react to the change in the top-level State

But comment indicates that we are not check State and Testing, so, seems it should be checked by caller.

> However, here you are comparing ctx.networkState and not ctx.deviceNetworkStatus.State - why is that?

Good point, it will not change the logic, but make it clear, modified.

Reply:
Ah - I looked that the wrong MostlyEqual() comment. But now I see there is a MostlyEqualStatus() for DeviceNetworkStatus which does check State and also Errors differing. Would it make sense to use that function?

Reply:
I do not think so, DeviceNetworkStatus do not check adresses, DNSServers or ProxyConfig changes, so it will have false-positive checks.